### PR TITLE
Update config to 0.12.0, actix-web to rc3

### PR DESCRIPTION
**Pull request by drconopoima**
Hi,

This is the same update that I created a pull request yesterday, they are just done on the chapter I'm still at, because the linking with clang fails and I couldn't verify my changes were compiling in your codebase

Interpreting the documentation for version 0.12.0 after the merge method got deprecated and it wasn't a straightforward upgrade took me a while, so I hope this to ease a bit of time.

Basically now the "try_into()" gets replaced by "try_deserialize" and the settings are replaced by ConfigBuilder

Hope it helps

### CODE REVIEW

**Comment by LukeMathWalker** [comment]
I'd prefer to wait on a new release of `config` to receive this change https://github.com/mehcode/config-rs/commit/e8985eacc7de0fb1916ab20a7bb40b9851c0f7da - otherwise all env variables need to be updated to be `APP__XXX` instead of `APP_XXX`.

**Comment by LukeMathWalker** [comment]
This has now been done on all branches, jumping to 0.13 directly!
